### ISSUE TITLE
IDE0054 "Use compound assignment" false positive in target-typed object initializer

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#49294]
**Version Used**:  16.8.0 and 16.9-p1

https://github.com/dotnet/roslyn/issues/33382 came back but only when using target-typed new.

There should be no IDE0054 diagnostics here:

```cs
class C
{
    public string P { get; set; }

    public C Clone()
    {
        //               ↓ IDE0054 Use compound assignment
        return new() { P = P + " (copy)" };
    }
}
```

**Comment by CyrusNajmabadi** [comment]
ickkkkkkkk
